src/alda/parser.clj

```diff
@@ -20,16 +20,12 @@
           :slur              (constantly :slur)
           :flat              (constantly :flat)
           :sharp             (constantly :sharp)
+          :natural           (constantly :natural)
           :dots              #(hash-map :dots (count %))
           :note-length       #(list* 'alda.lisp/note-length %&)
           :duration          #(list* 'alda.lisp/duration %&)
-          :pitch             (fn [s]
-                               (list* 'alda.lisp/pitch
-                                      (keyword (str (first s)))
-                                      (map #(case %
-                                              \- :flat
-                                              \+ :sharp)
-                                           (rest s))))
+          :pitch             #(list* 'alda.lisp/pitch %&)
+          :pitch-notation    #(list* 'alda.lisp/parse-pitch %)
```

Comment:
That should work -- I think it would be `#(parse-key-signature (apply parse-pitch %&))`

(`%&` is Clojure's special anonymous function syntax for "all the arguments, as a list," but the `parse-pitch` function takes a variable number of arguments (not a single list), so you can use `apply` to "unpack" the list. `parse-key-signature` takes a single list as an argument, so we don't need `apply` there)